### ISSUE TITLE
sending message to non-existent exchange also causes later messages to existing exchange to disappear

### ISSUE THREAD

**Issue by davidschneider** [SpinGo/op-rabbit#60]
here's a short test case

``` scala
    implicit val actorSystem = ActorSystem("rabbitactor")
    val rabbitControl = actorSystem.actorOf(Props[RabbitControl])

    Subscription.run(rabbitControl) {
      import Directives._
      channel() {
        consume(Binding.direct(Queue("somegoodqueue"), Exchange.direct("somegoodexchange"), List("somegoodroutingkey"))) {
          body(as[JsObject]) {
            data =>
              println(s"got $data")
              ack
          }
        }
      }
    }

    rabbitControl ! Message.exchange("hi", "somebadexchange", "somebadroutingkey")
    Thread.sleep(10000)

    for(i <- 1 to 10) rabbitControl ! Message.exchange(Json.obj("id" -> i), "somegoodexchange", "somegoodroutingkey")

    Thread.sleep(10000)
```

in this case, the 10 messages to the good exchange are never logged.  if i comment out the line where the message is sent to the bad exchange then all 10 good messages are logged

**Comment by timcharper** [comment]
Thank you for isolating this; do you have any idea why it's happening? I'm suspicious that the publishing actor itself is probably stopping due to an exception thrown, and the supervisor is not restarting it. In any case, the exception should be caught and handled properly, and this is awful!

I'll get your test incorporated in to the test suite.

**Comment by timcharper** [comment]
I looked in to this, looks like RabbitMQ disconnects the channel when a message is sent to an invalid exchange, and the publisher actor is getting stuck in a loop.

Working on a fix plus regression test.

**Comment by timcharper** [comment]
Fix will be released in 1.3.0

**Comment by timcharper** [comment]
Released
